maggma/stores.py

```diff
@@ -679,8 +693,17 @@ def update(self, docs, update_lu=True, key=None):
             if "_id" in search_doc:
                 del search_doc["_id"]
 
-            data = json.dumps(jsanitize(d)).encode("UTF-8")
-            self.collection.put(data, **search_doc)
+            if update_lu:
+                d[self.lu_field] = datetime.utcnow()
+
+            data = json.dumps(jsanitize(d["data"])).encode("UTF-8")
+            metadata = {self.lu_field: d[self.lu_field]}
+            metadata.update(search_doc)
+            self.collection.put(data, metadata=metadata)
+            self.transform_criteria(search_doc)
+            for fdoc in (self._files_collection.find(search_doc, ["_id"])
+                    .sort("uploadDate", -1).skip(1)):
+                self.collection.delete(fdoc["_id"])
```

Comment:
I think the alternative stores should still 'appear' to operate as MongoStores do. It shouldn't be necessary to know this is a GridFS store or writing builders will become obnoxious. Rather, this store should be given a set of keys, just like normal update that it will store in the metadata for searching: Maybe this is part of it's ensure_indexes ? 

GridFS has also been historically used with compression. This should be stored in the metadata as well. We tend to use "compression" : "zlib" and not much else.